### PR TITLE
Feature nixify

### PR DESCRIPTION
**Pull request by pjkundert**
Here's an updated elemental-chat that

- Uses a functionally defined holochain and cargo derivation from holo-nixpkgs, for improved build/test environment startup
- Uses the develop branches of holochain and hc-utils, w/ updated debug invocations
- Cleanup Makefile for improved target checking
- Works with tryorama 0.4.0-dev.0, with two Agents communicating within a single Conductor

### CODE REVIEW

**Comment by zippy** [comment]
Hey @pjkundert !  Thanks for this.  I cherry-picked your test fixes into an alpha1 release.  Please update your base for that, I just want some more review time on the nixification from others.

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,72 @@
+#
+# Test and build servicelogger Project
```
Comment (zippy) [inline Makefile]:
```suggestion
# Test and build elemental-chat
```

Comment (pjkundert) [reply Makefile]: Done, thanks!

**Comment by pjkundert** [comment]
I've rebased this on top of the current develop, and moved the pin of holochain repos to the top Cargo.lock.

**Comment by pjkundert** [comment]
Another option is to factor out the holo-nixpkgs rustc/cargo derivations and rust project build rules, and add them to holonix...  This would allow any holochain project to use the imperative rules currently in holonix, *or* the functional approach.

**Comment by zippy** [comment]
Hey @pjkundert we've just done a bit more work on everything, and I didn't take the time yet to finish reviewing your work, so now there's another rebase needed...  :-(

**Comment by zippy** [comment]
obsolete

**Pull request closed without merging**